### ISSUE TITLE
unit tests failing "Unsupported class file major version 63"

### ISSUE THREAD

**Issue by philburk** [google/oboe#1815]
REPRO
Follow instructions at https://github.com/google/oboe/blob/main/tests/README.md

    cd tests
    ./run_tests.sh

EXPECT test to run

ACTUAL
Build failed with:
```

Building test runner app

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/philburk/github/oboe/tests/UnitTestRunner/build.gradle'

* What went wrong:
Could not compile build file '/Users/philburk/github/oboe/tests/UnitTestRunner/build.gradle'.
> startup failed:
  General error during conversion: Unsupported class file major version 63
  
  java.lang.IllegalArgumentException: Unsupported class file major version 63
```

**Comment by philburk** [comment]
Strange. The fix in #1816 worked a few times.
Then I switched some branches around.
Now the same fix fails with:

```
Building test runner app

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'UnitTestRunner'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find com.android.tools.build:gradle:7.6.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/com/android/tools/build/gradle/7.6/gradle-7.6.pom
       - https://maven.google.com/com/android/tools/build/gradle/7.6/gradle-7.6.pom
     Required by:
         project :
```

**Comment by philburk** [comment]
I tried setting 

    export JAVA_HOME=/Users/philburk/Library/Java/JavaVirtualMachines/corretto-18.0.2/Contents/Home/

I have tried using gradle 7.6, 7.5 and 7.5.1.
They all fail with "Could not find com.android.tools.build:gradle:7.*"

**Comment by philburk** [comment]
I then tried Java 17.

    export JAVA_HOME=/Users/philburk/Library/Java/JavaVirtualMachines/corretto-17.0.8/Contents/Home/

That failed with "Could not find com.android.tools.build:gradle".

So I reverted back to Gradle 7.2.1 by locally reverting #1816.
That worked.
But it complains about 

```
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 33
This Android Gradle plugin (7.2.1) was tested up to compileSdk = 32
```

This issue seems definitely related to a mismatch in versioning between gradle and Java.
I think my problem started when I upgraded to Java 20.
My old gradle was not compatible.  So I upgraded to gradle 8.0.2.
That is not stable so I downgraded to gradle 7.6 which still failed.

The only combination that worked for is gradle 7.2.1 and Java 17.0.8.
